jalbe0076/overlook
# Feature/user login

### What is the feature?
- This PR will close #15 
- Added a user login page and fetch the user if the credentials are valid
- Adjusted some styling
- Added error handling to the API functions

### How you implemented the solution?
- Added a modal asking the users to login other elements of the webpage are hidden
- Removed the hardcoded user ID allowing for whatever user gets inputted to be logged in
- Allow the user to log out

### Does it impact any other area of the project?
- Users now need to log in before viewing the webpage

### Any changes in the UI (Screenshots)?
- New login page
![image](https://github.com/jalbe0076/overlook/assets/123283073/175eff5e-a8d5-41a0-8d15-6f51d5c8ee2a)
- Adjusted some styling on the dashboard page and on room information cards
![image](https://github.com/jalbe0076/overlook/assets/123283073/43f3ed69-3fc1-468b-b3ee-1dbaac446ba0)
![image](https://github.com/jalbe0076/overlook/assets/123283073/4a5b28cf-10a9-4959-b3da-37fcfcb4a2ec)

### How to test the feature (A test scenario or any new setup)?
- Run the local server and enter the username and password, under the user dropdown you can logout
- Username: customer50 => note that the number can be changed to login different users IE: customer9 (max 50)
- Password: overlook2021 (for all users)
